src/backend/utils/misc/queryjumble.c

```diff
@@ -597,6 +597,24 @@ JumbleExpr(JumbleState *jstate, Node *node)
 				JumbleExpr(jstate, rtfunc->funcexpr);
 			}
 			break;
+		case T_Integer:
+			{
+				APP_JUMB(intVal(node));
+			}
+			break;
+		case T_GroupingFunc:
```

Comment:
According to the [commit](https://github.com/postgres/postgres/commit/5fd9dfa5f50e4906c35133a414ebec5b6d518493#diff-6a2775ed8b21d7748929e715791290669412ccae2e643bee5afb49c819c30049R398-R404), T_GroupingFunc for some reason was not backported by the commit f7ff7c8ecae4ce7ab3b73fd46171cdaa457b3591, and it does not look quite right and is located in the wrong place.

Reply:
Reason why logic for T_GroupingFunc wasn't backported in not 100% clear.
But I assume the reason is following:

ADB-6 definition of GroupingFunc:
```
typedef struct GroupingFunc
{
	NodeTag   type;
	List     *args;  /* arguments provided in the query. */
	int       ngrpcols; /* the number of grouping attributes */
} GroupingFunc;
```
is different comparing with the one from Postgres and latest GPDB:
```
typedef struct GroupingFunc
{
	Expr		xpr;
	List	   *args;			/* arguments, not evaluated but kept for
								 * benefit of EXPLAIN etc. */
	List	   *refs;			/* ressortgrouprefs of arguments */
	List	   *cols;			/* actual column positions set by planner */
	Index		agglevelsup;	/* same as Aggref.agglevelsup */
	int			location;		/* token location */
} GroupingFunc;
```
ADB-6 GroupingFunc definition is incompatible with T_GroupingFunc handling in original commit (as it refers to 'refs' field).
Thus current implementation is different as GroupingFunc is different.
The code location was discussed with Andrey - he asked to place all delta right before "default" statement.